From 60a9880008fea56a69aa6e143d320f18ba2c3411 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 16 Mar 2022 04:01:41 +0200
Subject: [PATCH] [SW-56706] rom: spi dma api fix

A dedicated bit the the dma ctrl register needs to be set in order
to dma data onto the fw sram. This patch fixes the same in the dma
logic used to copy ppboot+pcie_phy image.

Change-Id: Ie684b2a254a8bedcf8817207c796e2763c5f6b65
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index d9448d4..e4cf6f5 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -224,7 +224,7 @@
 	}
 
 	err = dma_copy_image_v2(desc->img_size, desc->spi_offset,
-				desc->loadaddr, DMA_DEST_HBW, false);
+				desc->loadaddr, DMA_DEST_SRAM, false);
 	if (err != DMA_SUCCESS) {
 		if (err == DMA_INVALID_CRC_ERROR) {
 			security_report_img_status(cur_img, IMG_STATUS_CRC_FAIL);
